doc/dev/contributing.md

```diff
@@ -0,0 +1,126 @@
+are still available through changes of the source code repository.
+In contrast, bug tracking systems come and go, and cannot be relied upon to
+track information about a change for extended periods of time.
+
+### Commit Message Format
+
+Unlike the content of the commit message, the format is relatively easy to
+verify for correctness. Having the same standard format allows Git tools like
+git shortlog to extract information from the patches more easily.
+
+The first line of the commit comment is the commit summary of the change.
+Changes submitted to the DAOS master branch require a DAOS Jira ticket number
+at the beginning of the commit summary. A DAOS Jira ticket is one that begins
+with DAOS and is therefore part of the DAOS project within Jira.
+For patches to other projects, such as CaRT, a different JIRA project should be
+used (e.g. CART).
+
+The commit summary should also have a component: tag immediately following the
+Jira ticket number that indicates which DAOS subsystem that the commit is
+related to. Example DAOS subsystems relate to modules like: client, pool,
+container, object, vos, rdb; functional components like rebuild; or auxiliary
+components like build, tests, doc. This subsystem list is not exhaustive,
+but provides a good guideline for consistency.
+
+The commit summary line must be 62 characters or less, including the Jira
+ticket number and component tag, so that git shortlog and git format-patch
+can fit the summary onto a single line. The summary must be followed by a blank
+like. The rest of the comments should be wrapped to 70 columns or less.
+This allows for the first line to be used a subject in emails, and also for the
+entire body to be displayed using tools like git log or git shortlog in an 80
+column window.
+
+```
+DAOS-nnn component: short description of change under 62 columns
+
+The "component:" should be a lower-case single-word subsystem of the
+DAOS code that best encompasses the change being made.  Examples of
+components include modules: client, pool, container, object, vos, rdb;
+functional subsystems: recovery; and auxiliary areas: build, tests, docs.
+This list is not exhaustive, but is a guideline.
+
+The commit comment should contain a detailed explanation of changes
+being made.  This can be as long as you'd like.  Please give details
+of what problem was solved (including error messages or problems that
+were seen), a good high-level description of how it was solved, and
+which parts of the code were changed (including important functions
+that were changed, if this is useful to understand the patch, and
+for easier searching).  Wrap lines at/under 70 columns.
+
+Signed-off-by: Your Real Name <your_email@domain.name>
+```
+
+### The `Signed-off-by:` line
+
+The `Signed-off-by:` line asserts that you have permission to contribute the
+code to the project according to the Developer's Certificate of Origin.
+
+### Additional commit tags
```

Comment:
These are all legacy, from our use of gerrit aren't they?

Reply:
I don't think that this is really related to Gerrit. I guess we can do whatever we want there.